WcaOnRails/app/controllers/committee_positions_controller.rb

```diff
@@ -0,0 +1,72 @@
+class CommitteePositionsController < ApplicationController
+  before_action :authenticate_user!, except: [ :index, :show ]
+  before_action -> { redirect_unless_user(:can_manage_committees?) }
+
+  def index
+    @committee = Committee.find_by_slug(params[:committee_id])
+    @committee_positions = @committee.committee_positions
+  end
+
+  def new
+    @committee = Committee.find_by_slug(params[:committee_id])
```

Comment:
That sounds much better. For the committee_position I might revert to using the id, since I wanted the slug not to be unique. That way 2 different committees could have a team-leader slug. For the teams, and committees I think this will clean things up a lot.